bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -224,11 +276,24 @@ RepoModelImport::mesh_data_t RepoModelImport::createMeshRecord(
 			}
 		}
 
+		if (props->first == REPO_IMPORT_UV)
```

Comment:
Can we rewrite this to a switch case? (I know this isn't your fault, it's pre-established!)

It's a bit stupid that we have all these if statements, if props->first == REPO_IMPORT_UV is true, it's not going to be true for REPO_IMPORT_MATERIAL or any other of the ifs around here.

So it's better if it's if...else, or a switch statement. The latter is cleaner i think

Reply:
I have changed it to an if/else chain. Unfortunately C++ doesn't allow switching on a string (dangit...). Ideally we could write an enum for the various JSON properties, but as you suggested at the start of this ticket, I tried to keep the refactoring fairly light.